service/service.go

```diff
@@ -549,28 +578,34 @@ func (t *service) Subscribe(opts ...core.SubOption) core.Subscription {
 			filter[id] = struct{}{}
 		}
 	}
-	listener := &subscription{
-		l:  t.bus.Listen(),
-		ch: make(chan core.ThreadRecord),
-	}
+	channel := make(chan core.ThreadRecord)
 	go func() {
-		for i := range listener.l.Channel() {
-			r, ok := i.(*record)
-			if ok {
-				if len(filter) > 0 {
-					if _, ok := filter[r.threadID]; ok {
-						listener.ch <- r
+		listener := t.bus.Listen()
+		for {
+			select {
+			case <-ctx.Done():
+				listener.Discard()
+				close(channel)
```

Comment:
Should this two lines be _defered_ at first LOC of the goroutine to cover the other `case`, or isn't needed?

Reply:
Yep, good catch!